Scripts/SecretServer/Slack/Discovery/Slack-Discovery.ps1

```diff
@@ -1,37 +1,26 @@
-<#
-Slack User Role Definitions
-is_admin = Indicates whether the user is an Admin of the current workspace
-is_owner = Indicates whether the user is an Owner of the current workspace.
-is_restricted = Indicates whether or not the user is a guest user.
-is_ultra_restricted = Indicates whether the restricted user is a single-channel guest.
-Is_app_user = Indicates whether the user is an authorized user of the calling app.
-Is_bot = Indicates whether the user is actually a bot user. Bleep bloop. Note that Slackbot is special, so is_bot will be false for it.
-#>
-
 [string]$baseURL = "https://slack.com"
 [string]$api = "$baseURL/api"
 [string]$DiscoveryMode = $args[0] #Select "Default" or "Advanced"
-[string]$OAuthToken = $args[1]
+[string]$AccessToken = $args[1]
 [string]$adminrole = $args[2]
 [string]$svcacctrole = $args[3]
 
-
-$accessToken = $OauthToken
-
+if ($DiscoveryMode -notin "Default", "Advanced") { Throw "Invalid Discovery Mode" }
 
 #Script Constants
-[string]$LogFile = "$env:ProgramFiles\Thycotic Software Ltd\Distributed Engine\log\Slack-Discovery.log"
+[string]$logApplicationHeader = "Slack-Discovery"
+[string]$LogFile = "$env:ProgramFiles\Thycotic Software Ltd\Distributed Engine\log\$LogApplicationHeader.log"
```

Comment:
@jagger is the "$" at the end going to cause issues?

Reply:
no, this minimizes spots that could be missed if the integration is used as a template